bevy_lint/src/lints/panicking_methods.rs

```diff
@@ -0,0 +1,290 @@
+                    // If performance is an issue in the future, this could be cached.
+                    let key = Symbol::intern(panicking_method);
+
+                    if path.ident.name == key {
+                        // It is one of the panicking methods. Write down the alternative and stop
+                        // searching.
+                        break 'block *alternative_method;
+                    }
+                }
+
+                // If we reach this point, the method is not one we're searching for. In this case,
+                // we exit.
+                return;
+            };
+
+            // By this point, we've verified that `src` is a panicking type and the method is one
+            // that panics with a viable alternative. Let's emit the lint.
+
+            // Try to find the string representation of `src`. This usually returns `my_query`
+            // without the trailing `.`, so we manually append it. When the snippet cannot be
+            // found, we default to the qualified `Type::` form.
+            let src_snippet = snippet_opt(cx, src.span).map_or_else(
+                || format!("{}::", panicking_type.name()),
+                |mut s| {
+                    s.push('.');
+                    s
+                },
+            );
+
+            // Try to find the generic arguments of the method, if any exist. This can either
+            // evaluate to `""` or `"::<A, B, C>"`.
+            let generics_snippet = path
+                .args // Find the generic arguments of this path.
+                .and_then(GenericArgs::span_ext) // Find the span of the generics.
+                .and_then(|span| snippet_opt(cx, span)) // Extract the string, which may look like `<A, B>`.
+                .map(|snippet| format!("::{snippet}")) // Insert `::` before the string.
+                .unwrap_or_default(); // If any of the previous failed, return an empty string.
+
+            // Try to find the string representation of the arguments to our panicking method. See
+            // `span_args()` for more details on how this is done.
+            let args_snippet = snippet(cx, span_args(args), "");
+
+            span_lint_and_help(
+                cx,
+                panicking_type.lint(),
+                method_span,
+                format!(
+                    "called a `{}` method that can panic when a non-panicking alternative exists",
+                    panicking_type.name()
+                ),
+                None,
+                // This usually ends up looking like: `query.get_many([e1, e2])`.
+                format!("use `{src_snippet}{alternative}{generics_snippet}({args_snippet})` and handle the `Option` or `Result`"),
+            );
+        }
+    }
+}
+
+enum PanickingType {
+    Query,
+    QueryState,
+    World,
+}
+
+impl PanickingType {
+    /// Returns [`Self`] if the type matches Bevy's `Query` or `QueryState` types.
```

Comment:
Now also handles `World` type.
I think this comment will be prone to get outdated again in the future, perhaps we could rewrite it without mentioning the concrete types, or framing them as examples?